src/main/java/edu/unc/lib/boxc/migration/cdm/services/CdmIndexService.java

```diff
@@ -335,6 +360,52 @@ private String indexFieldType(String exportField) {
         }
     }
 
+    /**
+     * Indexes all exported objects for this project
+     * @param options
+     * @throws IOException
+     */
+    public void indexAllFromCsv(CdmIndexOptions options) throws IOException {
+        assertObjectsExported(options);
+
+        CdmFieldInfo fieldInfo = fieldService.loadFromCsvFieldsFromProject(project);
+        List<String> exportFields = fieldInfo.listAllExportFields();
+        outputLogger.info(exportFields.toString());
+        recordInsertSqlTemplate = makeInsertTemplate(exportFields);
+
+        try (
+                var conn = openDbConnection();
+                Reader reader = Files.newBufferedReader(options.getCsvFile());
+                CSVParser csvParser = new CSVParser(reader, CSVFormat.DEFAULT
+                        .withFirstRecordAsHeader()
+                        .withHeader(String.valueOf(exportFields))
+                        .withTrim());
+        ) {
+            for (CSVRecord csvRecord : csvParser) {
+                if (!csvRecord.get(0).isEmpty()) {
+                    List<String> fieldValues = new ArrayList<>();
+                    for (int i = 0; i < csvRecord.size(); i++) {
+                        fieldValues.add(csvRecord.get(i));
+                    }
+                    indexObject(conn, fieldValues);
+                }
+            }
+        } catch (IOException e) {
+            throw new MigrationException("Failed to read export files", e);
+        } catch (SQLException e) {
+            throw new MigrationException("Failed to update database", e);
+        }
+
+        project.getProjectProperties().setIndexedDate(Instant.now());
+        ProjectPropertiesSerialization.write(project);
+    }
+
+    private void assertObjectsExported(CdmIndexOptions options) {
```

Comment:
Since this isn't checking that objects are exported anymore, maybe `assertCsvImportValid` or `assertCsvImportExists` or something